src/interfaces/interfaces.ts

```diff
@@ -111,9 +111,7 @@ namespace interfaces {
         setCurrentRequest(request: Request): void;
     }
 
-    export interface ReflectResult {
```

Comment:
This was deleted as targetKey has changed to string | symbol.

Note that there was already
```
    export interface MetadataMap {
        [propertyNameOrArgumentIndex: string]: Metadata[];
    }
```

but !
```
export interface MetadataReader {
        //..
        getPropertiesMetadata(constructorFunc: Function): MetadataMap;
    }
class MetadataReader implements interfaces.MetadataReader {
    public getPropertiesMetadata(constructorFunc: Function): interfaces.MetadataMap {
        // User generated properties annotations
        const userGeneratedMetadata =  Reflect.getMetadata(METADATA_KEY.TAGGED_PROP, constructorFunc) || [];
        return userGeneratedMetadata;
    }

}

//decorator_utils
function _tagParameterOrProperty(
    metadataKey: string,
    annotationTarget: any,
    key: string | symbol,
    metadata: interfaces.MetadataOrMetadataArray,
) {
    const metadatas: interfaces.Metadata[] = _ensureNoMetadataKeyDuplicates(metadata);

    let paramsOrPropertiesMetadata:any = {};
    // read metadata if available
    if (Reflect.hasOwnMetadata(metadataKey, annotationTarget)) {
        paramsOrPropertiesMetadata = Reflect.getMetadata(metadataKey, annotationTarget);
    }

    let paramOrPropertyMetadata: interfaces.Metadata[] | undefined = paramsOrPropertiesMetadata[key];

    if (paramOrPropertyMetadata === undefined) {
        paramOrPropertyMetadata = [];
    } else {
        for (const m of paramOrPropertyMetadata) {
            if (metadatas.some(md => md.key === m.key)) {
                throw new Error(`${ERROR_MSGS.DUPLICATED_METADATA} ${m.key.toString()}`);
            }
        }
    }

    // set metadata
    paramOrPropertyMetadata.push(...metadatas);
    //********** paramOrPropertiesMetadata is not a MetadataMap !!!
    paramsOrPropertiesMetadata[key] = paramOrPropertyMetadata;
    Reflect.defineMetadata(metadataKey, paramsOrPropertiesMetadata, annotationTarget);

}
```

Something is amiss !

Reply:
The result of above is that in reflection_utils we do not get a Target for symbol properties with inject / multiInject decorator due to Object.keys.
```
function getClassPropsAsTargets(metadataReader: interfaces.MetadataReader, constructorFunc: Function, constructorName:string) {

    const classPropsMetadata = metadataReader.getPropertiesMetadata(constructorFunc);
    let targets: interfaces.Target[] = [];
    const keys = Object.keys(classPropsMetadata);
```

I will add another commit that permits this kind of injection.

Reply:
Latest commits
permit injection.... this provided the missing functionality.  If we could change the interface MetadataMap to have key string or symbol then we should !

move symbol tests to container as browser tests were failing - due to no IE11 support for symbol.

Reply:
Target interface and class now have additional identifier property

Reply:
I could fix #944 in here too ? There is a pr #1177 for it.

I suspect that multiInject should also accept LazyServiceIdentifier.
```
function multiInject(serviceIdentifier: interfaces.ServiceIdentifier<any>) {
  return createTaggedDecoratorInternal(new Metadata(METADATA_KEY.MULTI_INJECT_TAG, serviceIdentifier));
}
//vs
function inject(serviceIdentifier: ServiceIdentifierOrFunc) {
  return createTaggedDecoratorInternal(new Metadata(METADATA_KEY.INJECT_TAG, serviceIdentifier),target => {
    if (serviceIdentifier === undefined) {
      throw new Error(UNDEFINED_INJECT_ANNOTATION(target.name));
    }
  })
}
```

Reply:
Hi @tonyhallett I will review the PR in a couple of minutes :), I think we should keep PR as small as possible so I think we should review and progress #1177 instead